lib/Sema/CSSimplify.cpp

```diff
@@ -9841,6 +9825,18 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
 
   case FixKind::AllowArgumentTypeMismatch: {
     increaseScore(SK_Fix);
+
+    // If there are any other argument mismatches already detected for this
+    // call, we increase the score even higher so more argument fixes means
+    // less viable is the overload.
+    if (llvm::any_of(getFixes(), [&](const ConstraintFix *fix) {
+          auto *fixLocator = fix->getLocator();
+          return fixLocator->findLast<LocatorPathElt::ApplyArgToParam>()
+                     ? fixLocator->getAnchor() == locator.getAnchor()
+                     : false;
+        }))
+      increaseScore(SK_Fix);
```

Comment:
I think having more than one parameter type mismatch should have bigger impact, let's increase each new one by 5, because the more parameters mismatch the least likely it's that this is an expect overload choice... So we could set impact either to be a `2` or `5`, `recordFixes` now supports adding impact as a second parameter, so this code doesn't have to call `increaseScore` manually like this.

Reply:
Great! Just changing :)